views/js/qtiCommonRenderer/renderers/Math.js

```diff
@@ -1,14 +1,52 @@
-define(['tpl!taoQtiItem/qtiCommonRenderer/tpl/math', 'taoQtiItem/qtiCommonRenderer/helpers/container', 'mathJax', 'lodash'], function(tpl, containerHelper, MathJax, _){
+/*
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2014 (original work) Open Assessment Technlogies SA (under the project TAO-PRODUCT);
+ *
+ */
+
+/**
+ * Math common renderer
+ *
+ * @author Sam Sipasseuth <sam@taotesting.com>
+ * @author Bertrand Chevrier <bertrand@taotesting.com>
+ */
+define([
+    'lodash',
+    'core/promise',
+    'tpl!taoQtiItem/qtiCommonRenderer/tpl/math',
+    'taoQtiItem/qtiCommonRenderer/helpers/container',
+    'mathJax'
+], function(_, Promise, tpl, containerHelper, MathJax){
+    'use strict';
+
     return {
         qtiClass : 'math',
         template : tpl,
         getContainer : containerHelper.get,
-        render : function(math, data){
-            if(typeof(MathJax) !== 'undefined' && MathJax){
-                _.delay(function(){//defer execution fix some rendering issue in chrome
-                     MathJax.Hub.Queue(["Typeset", MathJax.Hub, containerHelper.get(math).parent()[0]]);
-                },60);
-            }
+        render : function render (math, data){
+            return new Promise(function(resolve, reject){
+                if(typeof(MathJax) !== 'undefined' && MathJax){
+                    _.delay(function(){//defer execution fix some rendering issue in chrome
```

Comment:
would be nice to replace the delay with a proper way to listen when the mathjax typesetting is ready ...
like that ? http://docs.mathjax.org/en/latest/signals.html